torch/utils/data/distributed.py

```diff
@@ -63,5 +75,10 @@ def __iter__(self):
     def __len__(self):
         return self.num_samples
```

Comment:
I read that `num_samples` is the length of the padded dataset, no? If so, I would expect to get `self._dataset_length` here then. Would this break BC?

Reply:
Thanks for the quick reply! Good catch. Current implementation was always returning padded length. I have tweaked it so it returns `self._dataset_length` when `padding=False`. This doesn't break BC because `padding=True` by default.